tests/e2e/tests/mixer/mixer_test.go

```diff
@@ -415,32 +443,81 @@ func applyReviewsRoutingRules(t *testing.T) {
 	time.Sleep(30 * time.Second)
 }
 
-func visitProductPage(retries int) error {
-	standby := 0
-	for i := 0; i <= retries; i++ {
-		time.Sleep(time.Duration(standby) * time.Second)
-		http.DefaultClient.Timeout = 1 * time.Minute
-		resp, err := http.Get(fmt.Sprintf("%s/productpage", tc.gateway))
+// checkProductPageDirect
+func checkProductPageDirect() {
+	glog.Info("checkProductPageDirect")
+	dumpURL("http://localhost:"+productPagePort+"/productpage", false)
+}
+
+// dumpMixerMetrics fetch metrics directly from mixer and dump them
+func dumpMixerMetrics() {
+	glog.Info("dumpMixerMetrics")
+	dumpURL("http://localhost:"+mixerMetricsPort+"/metrics", true)
+}
+
+func dumpURL(url string, dumpContents bool) {
+	clnt := &http.Client{
+		Timeout: 1 * time.Minute,
+	}
+	status, contents, err := get(clnt, url)
+	glog.Infof("%s ==> %d, error=<%v>", url, status, err)
+	if dumpContents {
+		glog.Infoln(contents)
+	}
+}
+
+func get(clnt *http.Client, url string, headerkv ...string) (status int, contents string, err error) {
```

Comment:
(not for this pr probably) we could use the fortio client here too probably (and make it dump the full headers when there is an error)

Reply:
That would be nice.